whenchanged/whenchanged.py

```diff
@@ -91,6 +91,7 @@ def run_command(self, thefile):
             print_message += '.' + now.strftime('%f') + ", running '" + ' '.join(self.command) + "'"
         if print_message:
             print('==> ' + print_message + ' <==')
+        os.environ.putenv('WHEN_CHANGED_FILE', thefile)
```

Comment:
This would modify the environment variable of the parent process. It would be cleaner to only set the environment variable for the process spawned by subprocess, e.g. https://stackoverflow.com/questions/2231227/python-subprocess-popen-with-a-modified-environment

Reply:
hi @joh 
thanks for your review.
The PR is updated with your suggestion, please let me know if something else should be changed.

Reply:
@joh,
may you please validate this PR first #74?
Then if this update is merged, I will rebase this one.
Thanks

Reply:
Hi @joh,
the PR is now rebased on the master branch and should be mergeable as is.
best regards